AEPIdentity/Sources/Identity.swift

```diff
@@ -33,7 +33,7 @@ import Foundation
             return
         }
 
-        let hitQueue = PersistentHitQueue(dataQueue: dataQueue, processor: IdentityHitProcessor(responseHandler: handleNetworkResponse(hit:responseData:)))
+        let hitQueue = PersistentHitQueue(dataQueue: dataQueue, processor: IdentityHitProcessor(responseHandler: handleNetworkResponse(hit:responseData:), isEdgeRegistered: isEdgeRegistered))
```

Comment:
I would prefer not to change the HitQueue for this use-case and handle everything in the Extension, this can allow us to remove this code in the future since it will not be used for Edge use-cases.

Reply:
The addition of `func shouldQueue(entity: DataEntity) -> Bool` to the hit queue has applications outside of the Edge use case, for example, an extension can decide not to queue hits if the latest configuration contains an opted-out privacy status. IMO, this is a clean and protocol-driven approach to determining if a hit should be queued or not rather than loading up conditional statements to determine if a hit should be queued.

If someone using the hit queue doesn't really _want_ the new functionality they can simplify just implement `shouldQueue` to return true each time. This kind of pattern is seen throughout iOS, for example, https://developer.apple.com/documentation/uikit/uiviewcontroller/1621502-shouldperformseguewithidentifier

```swift
func shouldQueue(entity: DataEntity) -> Bool {
   return true
}
```

Reply:
should we even not initialize hitQueue when we know edge is enabled?
The current codes work, but it'd be better if can refactor the code to a less intrusive way where we don't need to check `isEdgeRegistered` at several different places.

Reply:
We could probably de-init the hitQueue once we know Edge is present. One of the reasons I went with injecting the check into the hit queue was to make it easily testable. If we keep all the logic for de-init'ing the hitQueue in Identity.swift it is only really testable via functional tests (maybe that's ok). But I preferred to have a solution that would lend itself better to unit testing.
thoughts?

Reply:
@emdobrin I'm not sure it makes more sense to not initialize the DB if not used. How would we not initialize the DB in this case? The DB is initialized in the init for Identity, at that point we don't know definitively whether or not the Edge extension is present or not.

If we never initialize the DB it could miss the case where the customer migrates to Edge but has existing Identity hits in the queue that need to go out. The proposed solution allows for existing hits that have been queued before Edge to be sent out, while not allowing additional new hits to be queued if Edge is present.

Reply:
I agree with Jiabin here that we should only check the presence of Edge ext once when the processing starts instead of per event and potentially de-init the queue in that case since it will not be used. 
With regards to the previously queued hits, since this is an async process and Identity can process those hits after some new events were sent to Edge, we can run into inconsistent situations where the identifiers are overwritten, so I would prefer to sync whatever we have latest to Edge directly. I will look for a confirmation on this scenario though.

Reply:
I'm a little concerned to set the queue to nil given an async task could be running. Not sure if that would lead to any memory issues. I've added an optimization to stop calling `getSharedState` once we know Edge is present. We should get the best of both worlds.